SRC/app.py

```diff
@@ -67,16 +67,22 @@ def handle_app_mention_events(body: dict, say: slack_bolt.Say, logger: logging.L
         logging object
     """
     # ID for channel message received from
-    channel_id = body["event"]["channel"]
+    # Set thread timestamp as channel ID 
+    try:
+        channel_id = body["event"]["thread_ts"]
+    except KeyError:
+        channel_id = body["event"]["ts"]
     
     # If no chain exists for this channel, create new one
     if not channel_id in THREADS_DICT:
         build_chain(channel_id)
     
     # Store user_message and get bot response
     bot_message = add_chain_link(channel_id, body, loc="mentions")
-    # Send generated response back to Slack
-    say(bot_message)
+
+    # Send generated response back in a thread
+    thread_timestamp = body["event"]["ts"]
+    say(bot_message, thread_ts = thread_timestamp)
```

Comment:
You mentioned that you couldn't pass the `thread_timestamp` variable you created directly into the `say()` function.

I think that's because the second parameter of the function will be something else, so it can't receive the `ts` object, which explains why it does work if you set `thread_ts=thread_timestamp`. I think that's what you mentioned, may be wrong.

Reply:
Looks fine to me, we tested it together and seems to work well.

Reply:
Looks good to me as well. Seems like the code is functioning on my end too.